sql/parser: use %q for parse error

madelynnblue opened the pull request:
This prevents really weird error messages that are rendered on a
terminal when a `\r` is present. For example:

```
6' as type int: strconv.ParseInt: parsing "3\r6": invalid syntaxparse "test1" as INTEGER: could not parse '3
```

Which was supposed to be

```
pq: <omitted>: row 4: parse "test1" as INTEGER: could not parse '3
6' as type int: strconv.ParseInt: parsing "3\r6": invalid syntax
```

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/18095)
<!-- Reviewable:end -->

madelynnblue committed:
sql/parser: use %q for parse error

This prevents really weird error messages that are rendered on a
terminal when a `\r` is present. For example:

```
6' as type int: strconv.ParseInt: parsing "3\r6": invalid syntaxparse "test1" as INTEGER: could not parse '3
```

Which was supposed to be

```
pq: <omitted>: row 4: parse "test1" as INTEGER: could not parse '3
6' as type int: strconv.ParseInt: parsing "3\r6": invalid syntax
```
